### ISSUE TITLE
revap errors again ? 

### ISSUE THREAD

**Issue by judg393** [jtsage/FS22_ProductionInspector#11]
Hi JTS 

hope your well 

I've no idea what ive done but i appear to be having problems again with revamp and your Production Inspector ?

animals and silo seems to work but productions cause errors :(

![20221107224012_1](https://user-images.githubusercontent.com/55981682/200431115-281c72c5-19da-412b-965b-b13f28053e1c.jpg)
![20221105215612_1](https://user-images.githubusercontent.com/55981682/200431119-d304824f-14f9-44e0-8c5f-311724a559e5.jpg)
[log.txt](https://github.com/jtsage/FS22_ProductionInspector/files/9956091/log.txt)

**Comment by judg393** [comment]
I tried to roll back to the version you sent me last time i had issues but i'm still haveing issues :( possibly a revamp update?
strangley animal works fine

**Comment by jtsage** [comment]
I'll try and spend some time on this one today.  I've been spending far too much time in MP lately, and this isn't one of the ones we use, so it's a bit lower on my radar.

Thanks for letting me know.

**Comment by judg393** [comment]
ah no worries mate I wasnt sure if you were even supporting this still :/ 
Its doing it on Silver R too

**Comment by judg393** [comment]
Did you get anywhere with this? Anything I can do to help? Not a coder but happy to test things :)

**Comment by BullFlag** [comment]
@judg393   I did remove FS22_A_ProductionRevamp and that cleared the buggy behavior with Production Inspector...  No longer needed option "Show Outputs" set to NO.

**Comment by judg393** [comment]
Aye unfortunately I really like PR :/ I mean PI does work so I'll just be patient. 😀

**Comment by jtsage** [comment]
I have been trying to get back to this, I'll try and move it up my priority list a bit more.  Too many projects, not enough hours in the day :)

**Comment by judg393** [comment]
Of course sorry! I really appreciate your time :)

**Comment by jtsage** [comment]
[FS22_ProductionInspector.zip](https://github.com/jtsage/FS22_ProductionInspector/files/10367464/FS22_ProductionInspector.zip)

Give this version a shot, would ya?

**Comment by BullFlag** [comment]
I'll try it with the new Beta of Revamp also  Thanks 

![2023-01-07 16_44_36-Revamp Scriptmod - LS Mapping Team   LSMT Modding Team and 3 more pages - Person](https://user-images.githubusercontent.com/23723361/211175611-4e7fd65c-53da-4341-909a-1822991fb9e7.png)

**Comment by jtsage** [comment]
Cool.  I have several things on my "want to do" list with this - there were some massive improvements that got as far as simpleinspector, but not this, gonna deadicate most of tomorrow to them, and hopefully push a new version to modhub / release page here

**Comment by judg393** [comment]
Jts you are a legend ;) thanks dude it's working for me too

**Comment by jtsage** [comment]
If either of you get particularly bored, this is a beta of 1.2.0.0 - assuming I didn't break anything, it will get pushed to modhub in the next few days.

[FS22_ProductionInspector.zip](https://github.com/jtsage/FS22_ProductionInspector/files/10388742/FS22_ProductionInspector.zip)

**Comment by BullFlag** [comment]
Beta 1.2... All looks good Only Noticed   Silo item "Top Right under Clock" setting bit buggy,  Didn't take setting change I made , flip back to previous.  Tried saving each change no difference

**Comment by jtsage** [comment]
As it happens, none of the display mode settings worked.  Fixed in this one. 

[FS22_ProductionInspector.zip](https://github.com/jtsage/FS22_ProductionInspector/files/10397091/FS22_ProductionInspector.zip)

**Comment by jtsage** [comment]
Closing as the fixed version hit modHub about 20min ago.  Feel free to reopen if I missed something

**Comment by judg393** [comment]
![20230116204106_1](https://user-images.githubusercontent.com/55981682/212762633-aeb2115a-f9f8-4302-91e6-c0f9a17d5b0b.jpg)
[log.txt](https://github.com/jtsage/FS22_ProductionInspector/files/10429380/log.txt)


OK What am i doing wrong :(

**Comment by jtsage** [comment]
Hmm.  The folder switcher display is perplexing to me, lemme run down why that might happen. (Can you check to see if you have both the ZIP and an unzipped version in your collections somewhere (at a guess, in "The Oaks") - that might do it)  - The only time the "same" mod should be able to appear twice is if it is both zipped and unzipped.  I hope :)

As to the error and blank display - correct me if I'm wrong, but it *looks* like you have productions and silo both displaying in mode 1 (under the help menu).  Do you by chance not own any productions, or not have any working?  Or maybe not have any silos with anything in them?  It's getting hung up on bit that stacks the different display elements neatly, but before I go in and do a ton of checks to catch `nil` errors, I'd like to be able to reproduce locally.

Thanks!

**Comment by judg393** [comment]
The folder and zip thing is likely I was testing using GitHub's desktop app for mod updates. I'll check it.

Regarding PI I own 1 manure factory but it's not running anything and apparently I own a warehouse that's baked into map again not in use. 

I haven't messed with my positioning yet so yes it's likely PI and SI (I don't own or have anything stored in the map silos afaik but they do show up) should I try and change the positioning of thr Insp. windows 

I'm really sorry I feel like you put in all this work and I just break things 🙈

**Comment by judg393** [comment]
Ok I'm an idiot moved the windows around and bobs your uncle

**Comment by jtsage** [comment]
cool.  i've not yet circled back to this, but that is what I thought might be the issue.  There are some math operations in use where the value should always be defined, except apparently it is not.  I'll drop some extra checks to get it to not error on this, and perhaps I'll stumble across what is actually causing it.

**Comment by judg393** [comment]
Thank you and sorry to cause problems 🙈🤦‍♂️

**Comment by jtsage** [comment]
I threw in a few extra checks that should keep this from happening.  Also, I added unitConvert (or something close to it) to the mod.  It's the ZIP in the repository, gonna give it a go tonight and tomorrow before I push to modHub again.

**Comment by judg393** [comment]
Ooo what's unit convert 😂 I love how much you build into your mods I will tie a knot in my neck to grab latest and replace

**Comment by jtsage** [comment]
UnitConvertLite is/was a mod by ThundR in the 19 days, it has been ported over to 22 - some of his choices I like, and some I don't - most of those I don't were based on the limitations of 19, and I understand a new version is being worked on, looking forward to it.

But, unitConvert does just that - it'll display volume or weight in whatever unit you prefer.  Would you like to know how much wheat is in your trailer, in metric tons?  It'll do that.  How many gallons of water in the tanker?  It'll do that too.

And yes, I have not pushed a "real" release yet, probably today - usually this file : https://github.com/jtsage/FS22_ProductionInspector/blob/main/FS22_ProductionInspector.zip is the most recent.

**Comment by judg393** [comment]
Ok I'm using the new file I'll see what I find. I can see it's added a whole load of info to the bottom of my f1 menu. Hope it doesn't interfere with lumberjack.

**Comment by BullFlag** [comment]
![2023-01-27 08_28_53-Farming Simulator 22](https://user-images.githubusercontent.com/23723361/215146519-0ce525d4-13cf-470d-80dd-9e75117ee1cd.png)
The adding in the Unit Convert is a genius improvement.   Running that all the time now    With limited use all looking good with Beta 10 Revamp..

**Comment by judg393** [comment]
I agree no issues here though I'm gonna be honest not sure I'm gonna use unit convert aspect it's not broken anything (and I haven't either 😬)

**Comment by jtsage** [comment]
Haha.  I mostly agree with you there.  I don't have a good visual understanding of how much grain 1l is, but that number in bushels sure doesn't help me more.

**Comment by judg393** [comment]
Welp revamp updated yesterday and appears to have broken PI again ![image](https://user-images.githubusercontent.com/55981682/215702785-45b01a45-6ad0-49cf-8cca-8f45f6b8604c.jpeg)

**Comment by judg393** [comment]
thats forward planning can i just confirm what version i should be using i seemto have 1.2.0.1???

**Comment by jtsage** [comment]
Latest here is https://github.com/jtsage/FS22_ProductionInspector/releases/tag/v1.2.0.2

But if it's related to the outputs, 1.2.0.2 probably won't fix it.  I need to make the section a bit more bulletproof.

**Comment by judg393** [comment]
i know you'll fix what you can when

**Comment by jtsage** [comment]
[FS22_ProductionInspector.zip](https://github.com/jtsage/FS22_ProductionInspector/files/10563607/FS22_ProductionInspector.zip)

Give this one a shot.  Loaded up fine for me with a couple productions on a map and revamp installed, but who knows.  If it does it again, can you drop your game log here too?  Thanks!

**Comment by judg393** [comment]
Ooo just seen this will try now

**Comment by judg393** [comment]
We'll blow me whatever you did worked :) thanks mate

**Comment by FirenzeIT** [comment]
Yes, always errors for me too with (and sometimes without) revamps addon :-(
